docs/email/postfix/postfix-spf-dkim-debian-8.md

```diff
@@ -0,0 +1,375 @@
+
+The tags in between identify the servers mail for your domain can come from. "mx" is a shorthand for all the hosts listed in MX records for your domain. If you've got just one mail server, it's probably the best option. If you've got a backup mail server (a second MX record) using "mx" won't cause any problems, your backup mail server will be advertised as an authorized source for mail and it just won't ever send any. The "a" tag lets you identify a specific host by name or IP address, letting you be specific about exactly which hosts are authorized. You'd use it if you wanted to be pedantic about the backup mail server not being allowed to send outgoing mail, or if you wanted to identify hosts other than your own mail server that could send mail from your domain (eg. putting your ISP's outgoing mail servers in the list so they'd be recognized when you had to send mail through them). For now we're going to stick with the "mx" version, it's simpler and is correct for most basic configurations including ones that handle multiple domains.
+
+To add the record, go to your DNS management interface and add a record of type TXT for your domain itself (ie. a blank hostname) containing this string:
+
+    "v=spf1 mx -all"
+
+If you're using Linode's DNS Manager, go to the domain zone page for the domain you want to set SPF up for and add a new TXT record. The screen will look something like this when you've got it filled out:
+
+![Linode DNS manager add TXT record](/docs/assets/Postfix_SPF_TXT_record.png)
+
+If your DNS provider allows it (DNS Manager doesn't) you should also add a record of type SPF, filling it in the same way as you did the TXT record.
+
+## Adding the SPF policy agent to Postfix ##
+
+The Python SPF policy agent adds SPF policy checking to Postfix. The SPF record for the sender's domain for incoming mail will be checked and if it exists mail will be handled accordingly. I selected the Python version because it's more capable and more up-to-date than the rather basic variant written in Perl.
+
+You'll need to switch to a root shell using `su -` or `sudo -s` for this.
+
+1. If you are using SpamAssassin to do spam filtering, you may want to edit `/etc/postfix-policyd-spf-python/policyd-spf.conf` to change the `HELO_reject` and `Mail_From_reject` settings to `False`. That will cause the SPF policy agent to run it's tests and and add a message header with the results in it but _not_ reject any messages. You may also want to make this change if you want to see the results of the checks but not actual apply them to mail processing. Otherwise, just go with the standard settings.
+
+2. Edit `/etc/postfix/master.cf` and add the following entry at the end:
+
+    policyd-spf  unix  -       n       n       -       0       spawn
```

Comment:
Please use the [file-excerpt](https://linode.com/docs/linode-writers-formatting-guide#files-and-file-excerpts) tag to show snippets that belong in a file. This will also help show the correct indentation as it should be in the file.